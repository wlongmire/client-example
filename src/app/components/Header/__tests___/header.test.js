/* eslint-env node, jest */

import React from 'react'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import { Header } from '../index'

jest.mock('app/actions/userActions', () => 'testActions')

// Snapshot for Home Header Component
describe('>>> Header --- Snapshot', () => {
  beforeEach(() => {
    global.userPool = {}
    global.userPoolId = '123123123'
  })

  it('+++capturing Snapshot of Header without User', () => {
    const renderedValue = renderer.create(
      <Header
        user={null}
        logout={jest.fn()}
        resetForm={jest.fn()}
      />).toJSON()

    expect(renderedValue).toMatchSnapshot()
  })
})

// Snapshot for Home Header Component
describe('>>> Header --- Snapshot', () => {
  beforeEach(() => {
    global.userPool = {}
    global.userPoolId = '123123123'
  })

  it('+++capturing Snapshot of Header with User', () => {
    const renderedValue = renderer.create(
      <Header 
        user={initialState.user}
        logout={jest.fn()}
        resetForm={jest.fn()}
      />).toJSON()
    
    expect(renderedValue).toMatchSnapshot()
  })
})

const initialState = { user: {
  _id: '5807af4bbd9f641100e26138',
  username: 'betamarshuser',
  firstName: 'Marsh',
  lastName: 'Beta',
  role: 'user',
  _brokerId: '5807af2edcba0f490c718471',
  accountPending: false }
}

describe('>>> Header--- Shallow + passing the {store} directly', () => {
  let container

  beforeEach(() => {
    global.userPool = {}
    global.userPoolId = '123123123'

    container = shallow(
      <Header
        user={initialState.user}
        logout={jest.fn()}
        resetForm={jest.fn()}
      />)
  })

  it('+++ check Prop matches with initialState', () => {
    beforeEach(() => {
      global.userPool = {}
      global.userPoolId = '123123123'
    })

    const instance = container.instance()
    expect(instance.props.user.username).toEqual('betamarshuser')
  })

  it('+++ check Prop matches with initialState', () => {
    beforeEach(() => {
      global.userPool = {}
    })

    const instance = container.instance()
    expect(instance.props.user.lastName).toEqual('Beta')
  })
})
