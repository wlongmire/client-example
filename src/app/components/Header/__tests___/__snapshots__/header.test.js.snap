// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Header --- Snapshot +++capturing Snapshot of Header with User 1`] = `
<nav
  className="navbar navbar-inverse"
>
  <div
    className="container"
  >
    <div
      className=""
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="/submissions"
        >
          Owner's Edge
        </a>
        <button
          className="navbar-toggle collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
      </div>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
      >
        <ul
          className="nav navbar-nav navbar-right"
          role={null}
        >
          <li
            className="nav-link"
            role="presentation"
            style={undefined}
          >
            <a
              action="push"
              href="#"
              onClick={[Function]}
              role="button"
            >
              Submissions
            </a>
          </li>
          <li
            className="nav-link"
            role="presentation"
            style={undefined}
          >
            <a
              action="push"
              href="#"
              onClick={[Function]}
              onMouseUp={undefined}
              role="button"
            >
              Submit New
            </a>
          </li>
          <li
            className="nav-link"
            role="presentation"
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
            >
              Log Out
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</nav>
`;

exports[`>>> Header --- Snapshot +++capturing Snapshot of Header without User 1`] = `
<nav
  className="navbar navbar-inverse"
>
  <div
    className="container"
  >
    <div
      className=""
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="/submissions"
        >
          Owner's Edge
        </a>
        <button
          className="navbar-toggle collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
      </div>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
      >
        <ul
          className="nav navbar-nav navbar-right"
          role={null}
        >
          <li
            className="nav-link"
            role="presentation"
            style={undefined}
          >
            <a
              action="push"
              href="#"
              onClick={[Function]}
              role="button"
            >
              Log In
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</nav>
`;
